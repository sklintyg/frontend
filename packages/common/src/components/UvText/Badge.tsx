import * as React from 'react'
import styled from 'styled-components'

const Root = styled.div`
  overflow-wrap: anywhere;
  white-space: pre-wrap;

  ul {
    list-style: unset;
    padding-left: 15px;
  }
`

const Badge: React.FC = ({ children }) => {
  return <Root className={'ic-badge ic-badge--small iu-bg-secondary-light'}>{children}</Root>
}

export default Badge
