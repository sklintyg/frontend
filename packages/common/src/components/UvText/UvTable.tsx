import { ConfigViewColumn, ValueText, ValueTextRow } from '@frontend/common'
import React from 'react'

export interface Props {
  columns: ConfigViewColumn[]
  rows: ValueTextRow[]
}

const UvTable: React.FC<Props> = ({ columns, rows }) => {
  return (
    <table className="ic-table iu-fullwidth">
      <thead>
        <tr>
          {columns.map((column: ConfigViewColumn, index: number) => (
            <th key={index} scope="col">
              {column.text}
            </th>
          ))}
        </tr>
      </thead>
      {rows && (
        <tbody>
          {rows.map((row: ValueTextRow, index: number) => (
            <tr key={index}>
              {columns.map((column: ConfigViewColumn, index: number) => (
                <td key={index}>{row.columns.find((rowColumn: ValueText) => rowColumn.id === column.id)?.text ?? ''}</td>
              ))}
            </tr>
          ))}
        </tbody>
      )}
    </table>
  )
}

export default UvTable
