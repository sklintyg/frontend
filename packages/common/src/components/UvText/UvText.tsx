import * as React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { CertificateBooleanValue, CertificateDataElement, CertificateDataValueType, CertificateTextValue } from '@frontend/common'
import grey from '@material-ui/core/colors/grey'
import { Typography } from '@material-ui/core'

const useStyles = makeStyles((theme) => ({
  root: {
    whiteSpace: 'pre-wrap',
    background: grey[300],
    display: 'inline-block',
    borderRadius: '4px',
    padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
  },
  uvText: {
    fontSize: theme.typography.subtitle2.fontSize,
  },
}))

interface UvTextProps {
  question: CertificateDataElement
}

const UvText: React.FC<UvTextProps> = ({ question }) => {
  const classes = useStyles()

  let displayText = 'Ej angivet'

  switch (question.value.type) {
    case CertificateDataValueType.BOOLEAN:
      const booleanValue = question.value as CertificateBooleanValue
      if (booleanValue.selected !== null && question.visible) {
        displayText = booleanValue.selected ? booleanValue.selectedText : booleanValue.unselectedText
      }
      break
    case CertificateDataValueType.TEXT:
      const textValue = question.value as CertificateTextValue
      if (textValue.text != null && textValue.text.length > 0) {
        displayText = textValue.text
      }
      break
    default:
      displayText = 'Ok√§nd datatyp'
      break
  }

  return (
    <div className={classes.root}>
      <Typography className={classes.uvText}>{displayText}</Typography>
    </div>
  )
}

export default UvText
