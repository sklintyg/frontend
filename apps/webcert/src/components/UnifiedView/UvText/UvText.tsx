import type React from 'react'
import { Badge } from '../Badge'

export const UvText: React.FC<{ value: { text?: string | null } }> = ({ value }) => (
  <Badge>{typeof value.text === 'string' && value.text.length > 0 ? value.text : 'Ej angivet'}</Badge>
)
