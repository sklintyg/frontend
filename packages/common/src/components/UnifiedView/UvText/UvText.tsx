import React from 'react'
import { Badge } from '../Badge'
import { Value } from '../../../types/certificate'

export const UvText: React.FC<{ value: Value }> = ({ value }) => (
  <Badge label={typeof value.text === 'string' ? value.text : 'Ej angivet'} />
)
