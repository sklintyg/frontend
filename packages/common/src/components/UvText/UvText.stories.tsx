import React from 'react'

import { Story } from '@storybook/react'
import { fakeBooleanElement, fakeTextElement } from '../../utils/faker/fakeCertificateData'
import UvText, { Props } from './UvText'

export default {
  title: 'Components/UvText',
  component: UvText,
}

const Template: Story<Props> = (args) => <UvText {...args} />

export const BooleanValue = Template.bind({})
BooleanValue.args = {
  question: fakeBooleanElement({ id: '1' })['1'],
}

export const TextValue = Template.bind({})
TextValue.args = {
  question: fakeTextElement({ id: '1' })['1'],
}
