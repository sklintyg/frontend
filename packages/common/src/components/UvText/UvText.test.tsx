import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import UvText from './UvText'
import {
  CertificateBooleanValue,
  CertificateDataElement,
  CertificateDataValue,
  CertificateDataValueType,
  CertificateTextValue,
} from '@frontend/common'

describe('UvText', () => {
  it('renders without crashing', () => {
    const question = createQuestionWithTextValue()
    const div = document.createElement('div')
    ReactDOM.render(<UvText question={question} />, div)
  })

  it('displaying text value', () => {
    const question = createQuestionWithTextValue()
    const { getByText } = render(<UvText question={question} />)
    getByText(/Text/i)
  })

  it('displaying boolean value', () => {
    const question = createQuestionWithBooleanValue()
    const { getByText } = render(<UvText question={question} />)
    getByText(/Boolean value = true/i)
  })

  it('displaying empty value', () => {
    const question = createQuestion({ type: CertificateDataValueType.TEXT } as CertificateDataValue)
    const { getByText } = render(<UvText question={question} />)
    getByText(/Ej angivet/i)
  })

  it('displaying unknown value type', () => {
    const question = createQuestion({ type: CertificateDataValueType.UNKNOWN } as CertificateDataValue)
    const { getByText } = render(<UvText question={question} />)
    getByText(/OkÃ¤nd datatyp/i)
  })

  it('Verify snapshot', () => {
    const question = createQuestionWithTextValue()
    const tree = renderer.create(<UvText question={question} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

// Helper functions... Probably a good idea to create some utilities that can be reused....
export function createQuestionWithTextValue(): CertificateDataElement {
  const value: CertificateTextValue = {
    type: CertificateDataValueType.TEXT,
    text: 'Text',
    limit: 50,
  }
  return createQuestion(value)
}

export function createQuestionWithBooleanValue(): CertificateDataElement {
  const value: CertificateBooleanValue = {
    type: CertificateDataValueType.BOOLEAN,
    selected: true,
    selectedText: 'Boolean value = true',
    unselectedText: 'Boolean value = false',
  }
  return createQuestion(value)
}

export function createQuestion(value: CertificateDataValue): CertificateDataElement {
  return {
    id: 'id',
    readOnly: true,
    index: 0,
    mandatory: false,
    visible: true,
    parent: 'parent',
    validationErrors: [],
    validation: {
      hideExpression: 'null',
      required: false,
      requiredProp: 'null',
    },
    config: {
      text: 'Text',
      description: 'Description',
      component: 'null',
      prop: 'null',
    },
    value,
  }
}
