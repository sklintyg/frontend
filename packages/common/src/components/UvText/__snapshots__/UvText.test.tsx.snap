// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UvText Verify snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root makeStyles-uvText-2 MuiTypography-body1"
  >
    Text
  </p>
</div>
`;
